### ISSUE TITLE
Autoboot feature for a pre-selected ROM

### ISSUE THREAD

**Issue by mrehkopf** [mrehkopf/sd2snes#85]
Implement a feature to have the sd2snes boot a pre-selected ROM automatically on boot, without user interaction and without showing the menu.
Possible scenarios:
- Interactively select a file for subsequent auto-booting with a bail-out button combination to go back to the menu (more complex solution)
- Just expect a fixed filename (e.g. `autoboot.sfc`) and always boot that file if it exists (simple solution)

**Comment by MatthewCallis** [comment]
The second / simple solution leaves the user with no possible way to get out of that game without access to a machine that can read and manipulate the filesystem to change the name of the `autoboot.sfc` or remove it.

**Comment by mrehkopf** [comment]
To be fair they'd also need access to such a machine to put the file there in the first place though.

**Comment by MatthewCallis** [comment]
Sure, but going to an Expo or something where you want it to autoboot then its over and you just want free play you're stuck.

**Comment by magu** [comment]
With the bailout key combination, do you mean escaping from mid game out to menu or sequence to press while booting?

Simplest solution I can figure out would be just turning the SNES on with the A—or even just any—button pressed to clear the auto boot setting and revert to menu. Doesn't have to be any more sophisticated than that.

**Comment by Optiroc** [comment]
This is actually a feature I would want to use at home since I usually play just one title until I'm done with it, and skipping going through the menu at each "cold start" would be a win.

Here's my simple suggestion: Have a toggle in settings called "autoboot into latest file" (or preferably something more succinct! "autoboot", "kiosk mode", "kids mode"...). When that is set SD2SNES will always automatically load the last file except when holding a certain button combination. Preferably something easy enough to press using only the right hand so you simultaneously can press `reset` without using your foot. `R + X` or possibly `R + X + A` are my suggestions.

**Comment by ramapcsx2** [comment]
Simple solution is the autoboot.sfc + force enabled ingame key hooks + a memory resident flag that an autoboot game ran once. A user that enabled the feature is surely able to remember the back to menu key combo.
May I also suggest mapping the start button to acknowledge, so that starting the last loaded game is a simple double tap on start? This works great in Everdrives :)

**Comment by DieKatzchen** [comment]
I dunno, I like Magu's suggestion best. Forcing ingame hooks obviously prevents using this feature with any game that has trouble with ingame hooks. Nintendo has a long tradition of holding a button while booting, though.

**Comment by rarn** [comment]
Someone using autoboot will not go back to the menu often. In fact going back often means changing rom, which contradicts the need of autoboot. So I'd say there's no need for fancy bailout thing.
If the long reset could ignore the autoboot flag that would be the easiest, I think.

**Comment by dhincardoso** [comment]
Man! Implement the simple solution, it's enough for now for most of us, leave the complicated stuff for later, better have an option, even a simple one, than none. I absolutely agree with @rarn

**Comment by DhinCardoso** [comment]
Man! Implement the simple solution, it's enough for now for most of us, leave the complicated stuff for later, better have an option, even a simple one, than none. I absolutely agree with @rarn

**Comment by DhinCardoso** [comment]
@tandraschko why you referenced this request to the #86 and then referenced #86 back to #85? =/

**Comment by 2dcomplex** [comment]
It looks like it's been a couple years since any activity has been posted. Just wanted to chime in and say this would still be a great feature and am in full support of it getting implemented if at all possible.

**Comment by DhinCardoso** [comment]
> It looks like it's been a couple years since any activity has been posted. Just wanted to chime in and say this would still be a great feature and am in full support of it getting implemented if at all possible.

Agreed!

**Comment by CianMclennan** [comment]
Just to keep this alive. I would also like this feature.

**Comment by masterquest91** [comment]
2024 checkpoint. I would use this feature.

**Comment by AkimSALADIN** [comment]
Hello! I need this feature, please !

@mrehkopf, can you contact me via Twitter or Discord? I need your help creating this feature. 
Just give me the path to and I make the changes.

Basically, load "autoboot.sfc" located in the root of the storage and boot it directly without going through the SD2SNES menu.

Thank you !

**Comment by Ramsis-SNES** [comment]
> The second / simple solution leaves the user with no possible way to get out of that game without access to a machine that can read and manipulate the filesystem to change the name of the `autoboot.sfc` or remove it.

8 years later, I strongly agree with @MatthewCallis. :-)

**Comment by AkimSALADIN** [comment]
Yes, I know, but I need this functionality for personal purposes, so I would like to know where (in the source code) should go to add this feature please ?

Btw, It's just a suggestion but to avoid being stuck with the "autoboot.sfc" file, we can imagine doing a manipulation that allows you to skip the automatic loading of "autoboot.sfc", for example, start the console and hold the R button on SLOT1 (or SLOT2).
This would allow you to load the SD2SNES menu and then disable the feature in the menu.

**Comment by mektrix** [comment]
This would be useful for the Burn-in Test diagnostics rom, I'd love to use my sd2snes for this so I don't have to buy another cart.
